pkg/chaos/workflow.go

```diff
@@ -0,0 +1,338 @@
+package chaos
+
+import (
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	ymlparser "gopkg.in/yaml.v2"
+	"log"
+	"net/url"
+)
+
+type WorkflowYAML struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Name      string            `yaml:"name"`
+		Namespace string            `yaml:"namespace"`
+		Labels    map[string]string `yaml:"labels"`
+	} `yaml:"metadata"`
+	Spec struct {
+		Arguments       Arguments `yaml:"arguments"`
+		Entrypoint      string    `yaml:"entrypoint"`
+		SecurityContext struct {
+			RunAsNonRoot bool `yaml:"runAsNonRoot"`
+			RunAsUser    int  `yaml:"runAsUser"`
+		} `yaml:"securityContext"`
+		ServiceAccountName string     `yaml:"serviceAccountName"`
+		Templates          []Template `yaml:"templates"`
+	} `yaml:"spec"`
+}
+
+type Input struct {
+	Artifacts []Artifact `yaml:"artifacts"`
+}
+
+type Artifact struct {
+	Name string `yaml:"name"`
+	Path string `yaml:"path"`
+	Raw  Raw    `yaml:"raw"`
+}
+
+type Raw struct {
+	Data string `yaml:"data"`
+}
+
+type Template struct {
+	Name      string   `yaml:"name"`
+	Steps     [][]Step `yaml:"steps,omitempty"`
+	Input     Input    `yaml:"inputs,omitempty"`
+	Container struct {
+		Args    []string `yaml:"args"`
+		Command []string `yaml:"command"`
+		Image   string   `yaml:"image"`
+	} `yaml:"container,omitempty"`
+}
```

Comment:
Not all fields are considered specially metadata which might be needed in our case. ref: https://argoproj.github.io/argo/fields/#template

Reply:
Also take a look at the other structures and try making it match with the argo spec as workflows can be using some fields that aren't here and we would have to update these structs repeatedly.